src/deluge/gui/views/automation_instrument_clip_view.cpp

```diff
@@ -2710,81 +2722,69 @@ void AutomationInstrumentClipView::selectEncoderAction(int8_t offset) {
+			clip->lastSelectedParamArrayPosition = idx;
 		}
 
 		//if you're a synth or a kit (with affect entire off and a drum selected)
 		else if (instrument->type == InstrumentType::SYNTH
 		         || (instrument->type == InstrumentType::KIT && ((Kit*)instrument)->selectedDrum)) {
-
-			//if you haven't selected a parameter yet, start at the beginning of the list
-			if (isOnAutomationOverview()) {
-				auto idx = 0;
-				auto [kind, id] = nonKitAffectEntireParamsForAutomation[idx];
-				clip->lastSelectedParamID = id;
-				clip->lastSelectedParamKind = kind;
-				clip->lastSelectedParamArrayPosition = idx;
-			}
-			//if you are scrolling left and are at the beginning of the list, go to the end of the list
-			else if ((clip->lastSelectedParamArrayPosition + offset) < 0) {
-				auto idx = kNumNonKitAffectEntireParamsForAutomation - 1;
-				auto [kind, id] = nonKitAffectEntireParamsForAutomation[idx];
-				clip->lastSelectedParamID = id;
-				clip->lastSelectedParamKind = kind;
-				clip->lastSelectedParamArrayPosition = idx;
-			}
-			//if you are scrolling right and are at the end of the list, go to the beginning of the list
-			else if ((clip->lastSelectedParamArrayPosition + offset)
-			         > (kNumNonKitAffectEntireParamsForAutomation - 1)) {
-				auto idx = 0;
-				auto [kind, id] = nonKitAffectEntireParamsForAutomation[idx];
-				clip->lastSelectedParamID = id;
-				clip->lastSelectedParamKind = kind;
-				clip->lastSelectedParamArrayPosition = idx;
-			}
-			//otherwise scrolling left/right within the list
-			else {
-				auto idx = clip->lastSelectedParamArrayPosition + offset;
+			do {
+				//if you haven't selected a parameter yet, start at the beginning of the list
+				if (isOnAutomationOverview()) {
+					idx = 0;
+				}
+				//if you are scrolling left and are at the beginning of the list, go to the end of the list
+				else if ((clip->lastSelectedParamArrayPosition + offset) < 0) {
+					idx = kNumNonKitAffectEntireParamsForAutomation - 1;
+				}
+				//if you are scrolling right and are at the end of the list, go to the beginning of the list
+				else if ((clip->lastSelectedParamArrayPosition + offset)
+				         > (kNumNonKitAffectEntireParamsForAutomation - 1)) {
+					idx = 0;
+				}
+				//otherwise scrolling left/right within the list
+				else {
+					idx = clip->lastSelectedParamArrayPosition + offset;
+				}
 				auto [kind, id] = nonKitAffectEntireParamsForAutomation[idx];
-				clip->lastSelectedParamID = id;
-				clip->lastSelectedParamKind = kind;
-				clip->lastSelectedParamArrayPosition = idx;
-			}
+				if ((instrument->type == InstrumentType::KIT) && (id == Param::Unpatched::Sound::PORTAMENTO)) {
+					if (offset < 0) {
+						offset -= 1;
+					}
+					else if (offset > 0) {
+						offset += 1;
+					}
+				}
+				else {
+					clip->lastSelectedParamID = id;
+					clip->lastSelectedParamKind = kind;
+					clip->lastSelectedParamArrayPosition = idx;
+					break;
+				}
+			} while (true);
```

Comment:
A while true loop with a break is worse than a goto - you need to refactor so that the break condition is directly used as the loop condition

It might be easier to change the block above (getting the next param) into a separate function that can be called a second time when required

Reply:
I refactored into a separate function as suggested. Works nicely thanks!